core/src/main/scala/akka/persistence/cassandra/query/EventsByPersistenceIdStage.scala

```diff
@@ -329,6 +334,14 @@ import com.datastax.driver.core.utils.Bytes
                 if (pendingPoll.get >= seqNr)
                   query(switchPartition = false)
                 pendingPoll = None
+              } else if (pendingFastForward.isDefined) {
+                val nextNr = pendingFastForward.get
+                if (nextNr > seqNr) {
+                  internalFastForward(nextNr)
+                  query(switchPartition = false) // FIXME should we switch here?
```

Comment:
fast fw shouldn't trigger a query at all, it should only move the seqNr

hm, thinking of the order of `pendingPoll` and `pendingFastForward`
perhaps the order of this whole thing should be:
```
               if (empty && switchPartition && lookingForMissingSeqNr.isEmpty) {
                 if (refreshInterval.isEmpty) {
                   completeStage()
                 } else {
                   if (pendingFastForward.isDefined) {
                     val nextNr = pendingFastForward.get
                     if (nextNr > seqNr)
                       internalFastForward(nextNr)
                     pendingFastForward = None
                   }
                   if (pendingPoll.isDefined) {
                     if (pendingPoll.get >= seqNr)
                       query(switchPartition = false)
                     pendingPoll = None
                   }
                 }
               } else {
                  // TODO if we are far from the partition boundary we could skip this query if refreshInterval.nonEmpty
                  query(switchPartition = true) // next partition
               }

```

Reply:
Hmmm ok, I think I got it wrong and thought we always need to do that query when exhausting the previous query. I'll try this.